### PR TITLE
Update performance generation for new format

### PR DESCRIPTION
**Pull request by twicki**
## Purpose 
Updates the timing format to match https://github.com/VulcanClimateModeling/fv3core/pull/489

### CODE REVIEW

File: benchmarking/daint_single_node/stdout_to_json.py [hunk 1]
```diff
@@ -77,27 +61,43 @@ def stdout_to_json(stdout_file_regex, run_directory):
...
-            times[json_name]["maximum"] += raw_timers[fv3_name]["tmax"]
             times[json_name]["mean"] += raw_timers[fv3_name]["tavg"]
-
-    # assemble meta-data
+    for json_name in TIMER_MAPPING.keys():
+        times[json_name]["times"] = []
+        for rank in range(6):
+            times[json_name]["times"].append([])
+            for time_per_step in range(raw_timers["3.1-atmosphere_dynamics"]["hits"]):
+                times[json_name]["times"][rank].append(times[json_name]["mean"])
```
Comment (ofuhrer) [inline benchmarking/daint_single_node/stdout_to_json.py:90]:
```suggestion
            time_per_step = []
            for time_step in range(raw_timers["3.1-atmosphere_dynamics"]["hits"]):
                time_per_step.append(times[json_name]["mean"])
            times[json_name]["times"][rank] = time_per_step
```

File: benchmarking/daint_single_node/stdout_to_json.py [hunk 2]
```diff
@@ -77,27 +61,43 @@ def stdout_to_json(stdout_file_regex, run_directory):
...
                     times[json_name]["hits"] == raw_timers[fv3_name]["hits"]
                 ), "Can only accumulate timers with equal hit count"
-            times[json_name]["minimum"] += raw_timers[fv3_name]["tmin"]
-            times[json_name]["maximum"] += raw_timers[fv3_name]["tmax"]
             times[json_name]["mean"] += raw_timers[fv3_name]["tavg"]
-
-    # assemble meta-data
+    for json_name in TIMER_MAPPING.keys():
+        times[json_name]["times"] = []
+        for rank in range(6):
```
Comment (ofuhrer) [inline benchmarking/daint_single_node/stdout_to_json.py:87]: `NUM_MPI_RANKS = 6` in header?

File: benchmarking/daint_single_node/stdout_to_json.py [hunk 3]
```diff
@@ -77,27 +61,43 @@ def stdout_to_json(stdout_file_regex, run_directory):
...
             else:
                 assert (
                     times[json_name]["hits"] == raw_timers[fv3_name]["hits"]
                 ), "Can only accumulate timers with equal hit count"
-            times[json_name]["minimum"] += raw_timers[fv3_name]["tmin"]
-            times[json_name]["maximum"] += raw_timers[fv3_name]["tmax"]
             times[json_name]["mean"] += raw_timers[fv3_name]["tavg"]
-
-    # assemble meta-data
+    for json_name in TIMER_MAPPING.keys():
```
Comment (ofuhrer) [inline benchmarking/daint_single_node/stdout_to_json.py:85]:
```suggestion
    # mock data as if collected for every timestep
    for json_name in TIMER_MAPPING.keys():
```

File: benchmarking/daint_single_node/stdout_to_json.py [hunk 4]
```diff
@@ -77,27 +61,43 @@ def stdout_to_json(stdout_file_regex, run_directory):
...
+    return raw_timers
 
-    # convert into format for plotting
+
+def generate_output_from_times(
+    raw_timers: Dict[str, Dict[str, float]]
+) -> Dict[str, Any]:
+    """converts the raw data from fortran to the format used in fv3core's data collection"""
     times = {}
     for json_name, fv3_names in TIMER_MAPPING.items():
```
Comment (ofuhrer) [inline benchmarking/daint_single_node/stdout_to_json.py:75]:
```suggestion
    # extract mean timers over all timesteps
    for json_name, fv3_names in TIMER_MAPPING.items():
```

File: benchmarking/daint_single_node/stdout_to_json.py [hunk 5]
```diff
@@ -115,14 +115,54 @@ def stdout_to_json(stdout_file_regex, run_directory):
...
 
-    # print as JSON
+def print_to_output(setup: Dict[str, Any], times: Dict[str, Any], output=sys.stdout):
+    """Prints the collected data into the specified output"""
     experiment = {}
     experiment["setup"] = setup
     experiment["times"] = times
-    json.dump(experiment, sys.stdout, indent=4)
+    json.dump(experiment, output, indent=4)
     print("")
```
Comment (ofuhrer) [inline benchmarking/daint_single_node/stdout_to_json.py:127]:
```suggestion
```

File: benchmarking/daint_single_node/stdout_to_json.py [hunk 6]
```diff
@@ -115,14 +115,54 @@ def stdout_to_json(stdout_file_regex, run_directory):
...
+
+    match = extract_times_from_file(output_file)
+
+    raw_timers = parse_match_for_times(match)
+
+    times = generate_output_from_times(raw_timers)
+
+    setup = assemble_meta_data(output_file, run_directory, raw_timers)
+
+    print_to_output(setup, times)
```
Comment (ofuhrer) [inline benchmarking/daint_single_node/stdout_to_json.py:164]:
```suggestion
    output_file = find_output_file(run_directory, stdout_file_regex)
    match = extract_times_from_file(output_file)
    raw_timers = parse_match_for_times(match)
    times = generate_output_from_times(raw_timers)
    setup = assemble_meta_data(output_file, run_directory, raw_timers)
    print_to_output(setup, times)
```

**Review by ofuhrer — APPROVED** [review]
This is great. Sorry for taking so long with this. Some suggestions, mostly directly as code change suggestions. Nothing fundamental, so cherry-pick what you think is good input.

File: benchmarking/daint_single_node/stdout_to_json.py (hunk 5 above)
Comment (twicki) [reply benchmarking/daint_single_node/stdout_to_json.py:127]: This just make sure the HTML has a newline in the end of the report

**Pull request merged**